src/plugins/widgets/weather/WeatherSettings.tsx

```diff
@@ -1,64 +1,86 @@
-import React, { FC } from 'react';
+import React, { FC, useState, useEffect } from 'react';
 
+import { useDebounce } from '../../../hooks';
 import { Props, defaultData } from './types';
 import LocationInput from './LocationInput';
 
-const WeatherSettings: FC<Props> = ({ data = defaultData, setData }) => (
-  <div className="WeatherSettings">
-    <LocationInput
-      latitude={data.latitude}
-      longitude={data.longitude}
-      onChange={location => setData({ ...data, ...location })}
-    />
+const WeatherSettings: FC<Props> = ({ data = defaultData, setData }) => {
+  const [customName, setCustomName] = useState(data.name || '');
+  const debouncedName = useDebounce(customName, 500);
```

Comment:
You probably do not need to debounce the setting of the name if you don't refetch the forecast when the name changes.